<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1820</width>
    <height>980</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="_pointerData">
      <item>
       <widget class="QPlainTextEdit" name="_nonPolyPointerContent"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="_polyAsBasePointerContent"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="_polyAsLevel1PointerContent"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="_polyAsLevel2PointerContent"/>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="_pointerAddressData">
      <item>
       <widget class="QPlainTextEdit" name="_nonPolyPointerAddresses"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="_polyAsBasePointerAddresses"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="_polyAsLevel1PointerAddresses"/>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="_polyAsLevel2PointerAddresses"/>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="_buttonsLayout">
      <item>
       <widget class="QPushButton" name="_nonPolyInstantiateButton">
        <property name="text">
         <string>Instantiate (non-polymorphic)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="_polyAsBaseInstantiateButton">
        <property name="text">
         <string>Instantiate (polymorphic as base pointers)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="_polyAsLevel1InstantiateButton">
        <property name="text">
         <string>Instantiate (polymorphic as level1 pointers)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="_polyAsLevel2InstantiateButton">
        <property name="text">
         <string>Instantiate (polymorphic as a level2 pointer)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="_voidPtrInfoButton">
        <property name="text">
         <string>Void Ptr info</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="_nonSmartPtrExampleButton">
        <property name="text">
         <string>Non-SmartPointer Example</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="_clearBoxesButton">
        <property name="text">
         <string>Clear Boxes</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1820</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
